runtime/gc_base/modronapi.cpp

```diff
@@ -1089,6 +1092,30 @@ j9gc_notifyGCOfClassReplacement(J9VMThread *vmThread, J9Class *oldClass, J9Class
 	}
 }
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+void
+j9gc_prepare_for_checkpoint(J9VMThread *vmThread)
+{
+	MM_EnvironmentBase *env = MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+	MM_GCExtensionsBase *extensions = env->getExtensions();
+
+	/* trigger a GC to disclaim memory */
+	j9gc_modron_global_collect_with_overrides(vmThread, J9MMCONSTANT_EXPLICIT_GC_SYSTEM_GC);
+	j9gc_modron_global_collect_with_overrides(vmThread, J9MMCONSTANT_EXPLICIT_GC_PREPARE_FOR_CHECKPOINT);
+
+	extensions->configuration->adjustGCThreadCountOnCheckpoint(env);
+}
+
+BOOLEAN
+j9gc_reinitialize_for_restore(J9VMThread *vmThread)
+{
+	MM_EnvironmentBase *env = MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+	MM_GCExtensionsBase *extensions = env->getExtensions();
+
+	return extensions->configuration->reinitializeGCThreadCountOnRestore(env);
```

Comment:
inconsistent naming....
Caller is ending with _for_restore/_for_checkpoint
Callee is ending with OnRestore/OnCheckpoint

and another reasonable candidate would be 'after restore' (and 'before checkpoint')

Don't have strong preference, but would be nice to be consistent (now and in future).

Reply:
@amicic and I discussed that we'll leave this for later and make changes in the future with the next set of OMR changes